src/cogent3/app/__init__.py

```diff
@@ -33,24 +33,23 @@ def _get_app_attr(name, is_composable):
         name,
         is_composable,
         obj.__doc__,
-        _types["_data_types"],
-        _types["_return_types"],
+        ", ".join(sorted(_types["_data_types"])),
+        ", ".join(sorted(_types["_return_types"])),
     ]
 
 
 def available_apps():
     """returns Table listing the available apps"""
+    import cogent3.app as apps
+
     from cogent3.util.table import Table
 
     from .composable import Composable, __app_registry, user_function
 
-    rows = [_get_app_attr(app, True) for app in __app_registry]
+    # registration of apps does not happen until their modules are imported
+    for name in __all__:
+        importlib.import_module(f"cogent3.app.{name}")
 
-    mod = importlib.import_module(f"{__name__}.composable")
-    rows.append(
-        _get_app_attr(
-            f"{mod.__name__}.user_function", issubclass(user_function, Composable)
-        )
-    )
-    header = ["module", "name", "composable", "doc", "outputs", "data type"]
-    return Table(header, rows)
+    rows = [_get_app_attr(app, __app_registry.get(app)) for app in __app_registry]
```

Comment:
In `__app_registry`, keys are `get_object_provenance(klass)`. But here we use `app` as key. Are these equal?

Reply:
I'm iterating over the keys in the dict, so yes.

Reply:
Actually, there's no need to do the `__app_registry.get(app)`. The value must exist! Fixing that now.

Reply:
We have used `get` function for dict a lot of times, so we forget to use normal usage of key indexing.